src/shims/foreign_items.rs

```diff
@@ -243,14 +238,17 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 | "exit"
                 | "ExitProcess"
                 => {
-                    this.check_abi(abi, if link_name == "exit" { Abi::C { unwind: false } } else { Abi::System { unwind: false } })?;
-                    let &[ref code] = check_arg_count(args)?;
+                    let &[ref code] = this.check_shim(abi, if link_name == "exit" { Abi::C { unwind: false } } else { Abi::System { unwind: false } }, link_name_sym, args)?;
                     // it's really u32 for ExitProcess, but we have to put it into the `Exit` variant anyway
                     let code = this.read_scalar(code)?.to_i32()?;
                     throw_machine_stop!(TerminationInfo::Exit(code.into()));
                 }
                 "abort" => {
-                    this.check_abi(abi, Abi::C { unwind: false })?;
+                    this.check_abi_and_shim_symbol_clash(
+                        abi,
+                        Abi::C { unwind: false },
+                        link_name_sym,
+                    )?;
```

Comment:
This should also use `check_shim`, shouldn't it? The fact that it does not currently call `check_arg_count` seems like an oversight.